node/primitives/src/lib.rs

```diff
@@ -52,8 +52,10 @@ pub use disputes::{
 	SignedDisputeStatement, UncheckedDisputeMessage, ValidDisputeVote,
 };
 
-// For a 16-ary Merkle Prefix Trie, we can expect at most 16 32-byte hashes per node.
-const MERKLE_NODE_MAX_SIZE: usize = 512;
+// For a 16-ary Merkle Prefix Trie, we can expect at most 16 32-byte hashes per node
+// plus some overhead:
+// header 1 + bitmap 2 + partial_key 32 + children 16 * (32 + len 1) + value 32 + value len 1
```

Comment:
Actually small mistake, key len is u32 compact so max is more < 5 bytes, but having margin looks fine to me.